05-html-css/css-framework.md

```diff
@@ -37,111 +40,162 @@ Think of small as a very narrow viewport, like your phone horizontally. Medium w
+Another set of pre-defined classes are used to create alert panels calling a user's attention to notifications or error messages.   This is commonly used for flash notices or validation errors on forms.
 
-## Further Customization
-It is typical to use Sass when developing with Foundation. Using Sass allows you to customize easily by setting variables, yes variables in CSS! It's amazing!
+```html
+<section class="alert callout">
+  <h5>This is an alert callout</h5>
+  <p>It has an easy to override visual style, and is appropriately subdued.</p>
+  <a href="#">There is an error here!</a>
+</section>
+```
 
-But we're not there yet.
+![alert callout screenshot](imgs/alert-calloutv2.png)
 
-Instead you can overwrite Foundations styles your own CSS files. You can also set some customizations before downloading with [this page](http://foundation.zurb.com/sites/download.html/).
+You can experiment with callouts [here](https://codepen.io/adadev/pen/dmQodr?editors=1000).
 
-**Note:** There is a rails-foundation gem that seamlessly incorporates foundation into a rails application. The gem has the assumption that you want to utilize most of foundation’s functionality and that you know Sass. It is not recommended to use this gem quite yet. It includes a lot more of foundation’s functionality that we have not covered and may be overwhelming while you’re working on a new project.  But hey, if you’re feeling adventurous, go for it!
+**Experiment**:  Change the color of the callout background.
 
+## Form Validations
 
-## Add Foundation to your project
-We will be taking a version of Foundation from the web and installing in into a Rails application.
+Foundation also has a library named [**Abide**](https://foundation.zurb.com/sites/docs/abide.html) which can be used to go beyond the built-in HTML5 form validation available to us.  To use Abide over HTML5 validations, add `novalidate` and `data-abide` attributes to the form element.  
 
-1.  We can [download this specific version](https://cdnjs.cloudflare.com/ajax/libs/foundation/6.3.1/css/foundation.css) on this site. Note that the text loaded here is the CSS we will need so go ahead and select it all and copy it. Foundation made some breaking changes to a later version of Foundation, and so we'll be using an earlier version to ensure our usage of the grid layout is good.
+Elements can have a number of validations which ensure input matches common patterns like `alpha` (only letters), `alpha_numeric`, `color`, `cvv` (the three numbers on the back of a credit card), `date`, `datetime`, `day_month_year`, `domain`, `email`, `integer`, `month_day_year`, `number`, `time`, and `url`.  A pattern can be specified by adding a pattern attribute-value to the form field.  For example `<input type="text" pattern="email" >`.  Then when the user attempts to submit the form, Abide will check to ensure the field matches the given pattern.  
```

Comment:
I feel like the example is a bit buried in this wall of text. Could we just move it to its own line so that it is easier to find?